### PR TITLE
wire: add FAQ section

### LINKED ISSUE #555: wire: add an FAQ section to docs
After the Wire blog post and a variety of talks, there are some questions that come up rather regularly:

- How does Wire relate to other DI tools?
- Why is Wire part of Go Cloud?
- Why are the Wire directives a marker function package instead of comments?
- How should I deal with two dependencies of the same type?
- Who is using Wire?
- Should I use Wire for small applications?

These are exactly the same questions that the team has worked through over the design, and we have much crisper answers now than we used to, so we should write the answers down.

/cc @enocom

### PR DESCRIPTION
**Pull request by zombiezen**
Fixes #555
Updates #513

### CODE REVIEW

File: wire/README.md [hunk 1]
```diff
@@ -491,3 +491,116 @@ Create a new struct that includes the app plus all of the dependencies you want
...
+Follow [#513][] for updates.
+
+[#513]: https://github.com/google/go-cloud/issues/513
+
+### Why use pseudo-functions to create provider sets or injectors?
+
+In the early prototypes, Wire directives were specially formatted comments. This
+seemed appealing at first glance as this meant no compile-time or runtime
+impact. However, this unstructured approach becomes opaque to other tooling not
+written for Wire. Consider tools like [`gorename`][] or [guru][]: it would not
```
Comment (vangent) [inline wire/README.md]: "Tools like ... would not..."

File: wire/README.md [hunk 2]
```diff
@@ -491,3 +491,116 @@ Create a new struct that includes the app plus all of the dependencies you want
...
+
+### Should I use Wire for small applications?
+
+Probably not. Wire is designed to automate more intricate setup code found in
+larger applications. For small applications, hand-wiring dependencies is
+simpler.
+
+### Who is using Wire?
+
+Wire is still fairly new and doesn't have large users yet. However, we have
```
Comment (vangent) [inline wire/README.md]: large users -> a large user base ?

File: wire/README.md (hunk 1 above)
Comment (vangent) [inline wire/README.md]: Is there a link to the blog post yet? Consider adding one somewhere in the README....
https://blog.golang.org/wire

File: wire/README.md (hunk 2 above)
Comment (zombiezen) [reply wire/README.md]: Done.

File: wire/README.md (hunk 1 above)
Comment (zombiezen) [reply wire/README.md]: Added one at the top.

Comment (zombiezen) [reply wire/README.md]: Done.

File: wire/README.md [hunk 3]
```diff
@@ -491,3 +494,115 @@ Create a new struct that includes the app plus all of the dependencies you want
 to mock. Create a test-only injector that returns this struct, give it providers
 for the concrete mock types, and use `wire.Bind` to tell Wire that the
 concrete mock types should be used to fulfill the appropriate interface.
+
+## Frequently Asked Questions
+
+### How does Wire relate to other Go dependency injection tools?
+
+Other dependency injection tools for Go like [dig][] or [facebookgo/inject][]
+are based on reflection. Wire run as a code generator, which means that the
```
Comment (enocom) [inline wire/README.md]: s/run/runs/

File: wire/README.md [hunk 4]
```diff
@@ -491,3 +494,115 @@ Create a new struct that includes the app plus all of the dependencies you want
...
+
+[Dagger 2]: https://google.github.io/dagger/
+
+### Why is Wire part of Go Cloud?
+
+Wire was designed to reduce the toil in setting up the multitude of dependencies
+often found when interacting with common cloud providers. However, Wire is not
+coupled to Go Cloud or cloud servers: Wire is a standalone tool.
+
+The Go Cloud team is investigating splitting Wire into its own repository.
```
Comment (enocom) [inline wire/README.md]: nit: splitting out?

File: wire/README.md (hunk 3 above)
Comment (zombiezen) [reply wire/README.md]: Done.

File: wire/README.md (hunk 4 above)
Comment (zombiezen) [reply wire/README.md]: Done.

**Pull request merged**